app/controllers/map_controller.rb

```diff
@@ -16,7 +16,10 @@ def index
   end
 
   def show
-    @node = DrupalNode.find_map_by_slug(params[:name]+'/'+params[:date])
+    @node = DrupalNode.find params[:name]
```

Comment:
I'm sorry, Ananyo -- i think it'll help to refer to things as they are now
vs. things as they are in this pull request. The word "slug" had been used
in our codebase already to mean a string, used within a longer URL, where
the slug does not have to be unique. So let's call that "legacy slug" vs.
"friendly_id slug". What I mean is that we have firmly established
conventions for URLs already on the site, and the cost of changing those
is, I believe, too high. So I'm trying to think of ways that we could
maintain our existing URL schema, where parameterized "map name" ("New
York" => "new-york") is followed by the date, separated by a slash, as
shown in the routes.rb file:
https://github.com/publiclab/plots2/blob/master/config/routes.rb#L175

I'm sure you see that changing URL schemas is quite a serious proposal, and
we should do all we can to avoid this, because it'd result in either a)
broken links to legacy pages, or b) different URLs for legacy vs. new pages
of a given type (map).

I know this is frustrating, and I thank you for your patience and
thoroughness. I think we have to consider all creative possibilities for
adapting friendly_id to our legacy URL schema as shown in the current
(master branch) routes.rb before we turn to the alternative -- making a
proposal to the community of changing the legacy URL schema and either
redirecting or simply allowing old map URLs to break. Make sense? Thanks
again for your careful consideration!

On Mon, Jul 25, 2016 at 2:55 PM, Ananya Maiti <notifications@github.com>
wrote:

> In app/controllers/map_controller.rb
> <https://github.com/publiclab/plots2/pull/600#discussion_r72122575>:
>
> > @@ -16,7 +16,10 @@ def index
> >    end
> >
> >    def show
> > -    @node = DrupalNode.find_map_by_slug(params[:name]+'/'+params[:date])
> > +    @node = DrupalNode.find params[:name]
>
> Uh ok I think there is a bit misunderstanding in this part. The
> params[:name] basically contains the slug of the map not just the name.
> So the query searches for the map based on the friendly_id slug. So if
> there is a map with a similar name but a different date it will go as
> /map/NewYork-map--2/07-22-2016. So you see there are no two slugs which
> are same.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/600/files/c968e35c5fa7b3232e5e5b2ca0bc9afa658586cd#r72122575>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ0Cn7SniRJeXN0e_qUFwp88rIbuIks5qZQaNgaJpZM4JDL2B>
> .
>